backend/game/deck/deck_handler.py

```diff
@@ -58,10 +59,15 @@ def shuffle_discard_pile_to_draw_pile(self):
             self.__draw_pile.append(self.discard_pile.draw_from())
         self.__discard_pile.clear()
 
-    def buy_card_from_store(self, index):
-        bought = self.__card_store.pop(index)
-        self.__fill_card_store()
-        return bought
+    def buy_card_from_store(self, index, purchasing_player: Player):
+        card_to_buy: Card = self.__card_store[index]
+        if purchasing_player.energy < card_to_buy.cost:
```

Comment:
Stated this in a similar review, maybe best to have this check sit outside this method, but could see an argument for either way